### PR TITLE
[JENKINS-35423] CLI commands quiet-down/cancel-quiet-down extracted from core to CLI

### PR DESCRIPTION
**Pull request by pjanouse**
quiet-down and cancel-quiet-down covered by test-cases

Please review, thx. Pavel

### CODE REVIEW

File: test/src/test/java/hudson/model/QueueTest.java [hunk 1]
```diff
@@ -495,10 +495,10 @@ public void upstreamDownstreamCycle() throws Exception {
         assertThat("The cycle should have been defanged and chain3 executed", queue.getItem(chain3), nullValue());
     }
 
-    private static class TestFlyweightTask extends TestTask implements Queue.FlyweightTask {
+    public static class TestFlyweightTask extends TestTask implements Queue.FlyweightTask {
```
Comment (oleg-nenashev) [inline test/src/test/java/hudson/model/QueueTest.java]: Maybe better to move it to the Jenkins Test Harness lib

**Comment by oleg-nenashev** [comment]
Test failure is unlikely related

File: core/src/main/java/jenkins/model/Jenkins.java [hunk 2]
```diff
@@ -3458,11 +3458,14 @@ public synchronized HttpRedirect doQuietDown() throws IOException {
...
+     * Quiet down Jenkins - preparation for a restart
+     *
+     * @param block Block until the system really quiets down and no builds are running
+     * @param timeout If non-zero, only block up to the specified number of milliseconds
+     */
     @RequirePOST
-    public HttpRedirect doQuietDown(
-            @Option(name="-block",usage="Block until the system really quiets down and no builds are running") @QueryParameter boolean block,
-            @Option(name="-timeout",usage="If non-zero, only block up to the specified number of milliseconds") @QueryParameter int timeout) throws InterruptedException, IOException {
+    public HttpRedirect doQuietDown(boolean block, int timeout) throws InterruptedException, IOException {
```
Comment (olivergondza) [inline core/src/main/java/jenkins/model/Jenkins.java]: Is it safe to remove the `@QueryParameter` annotations?

Comment (oleg-nenashev) [inline core/src/main/java/jenkins/model/Jenkins.java]: I would vote against it

Comment (pjanouse) [inline core/src/main/java/jenkins/model/Jenkins.java]: OK, I'll remove both @Option, but leave @QueryParameter untouched.

**Comment by oleg-nenashev** [comment]
Retriggering the build after https://github.com/jenkinsci/jenkins/pull/2435

**Comment by daniel-beck** [comment]
@pjanouse Was this deliberately not `@Restricted`, or accidentally?

**Comment by olivergondza** [comment]
@daniel-beck, what do you mean, the old methods or new classes? If those methods are used fo UI only, I am ok to deprecate it instead so people are advised to stop using those. If the command classes should be restricted is something to be discussed - it was not done in the past though so there is a lot of relatively new unrestricted command classes.

**Comment by daniel-beck** [comment]
@olivergondza The other ones probably mostly predate `@Restricted`, and a lot of commands were added over the last year or so by @pjanouse, and I noticed today none are `@Restricted`. But I see no use for these commands in plugins. If we ever change signatures in those, we'd be in trouble. So I'd annotate them now to limit use in plugins.

**Comment by pjanouse** [comment]
@daniel-beck Mentioned commands are in the core for some time, I've been unbundling them only from the core to separate classes (as a preparation work to make a module from them), none of commands (= exposed a new API by this way) was added by me over the time.

On the other hand, from my PoV the CLI is one of the ways how user/admin can manage his deployment (other main ways are REST-API and WebGUI). Of course, it isn't 1:1 mapping, nor REST or WebGUI is. So from this PoV I don't think it is a good idea to restrict its usage only from shell scripts. Why sb. in the future can't depend on a module (created from these CLI cmds) and if he follows current approach (implemented in `CLICommand.main(...)`) he can use a CLI command programmaticaly? 

I'm not to add  `@Restricted` around them and so limit their usage.

**Comment by daniel-beck** [comment]
@pjanouse Groovy scripts ignore `@Restricted`, so typical administrative scripts are safe, as are all the intended usage of CLI commands. But we basically promise to keep plugin-accessible API backward compatible, and that seems like a burden for no benefit for CLI command classes.

We've seen in the past that plugin rely on `doSomething(…)` methods, when they should not have been, and when they changed arguments for Stapler reasons, broke. There's no reason for plugins to use these UI only methods, and I see CLI command implementations in the same realm.

**Comment by pjanouse** [comment]
[JENKINS-31098](https://issues.jenkins-ci.org/browse/JENKINS-31098) should be mentioned here. I'm not sure if it is the only one exception.

**Comment by olivergondza** [comment]
Since there is a way to invoke CLI commands from Java code[1] (even remotely), I guess we can restrict the classes. This will effectively force those clients to depend on the *CLI observable* aspects of the API (which we need to keep compatible for CLI clients) and not Java api that happens to be marked public (which we might want to change some day).

[1] https://github.com/olivergondza/jenkins-config-cloner/blob/90ed64949da7784960ca58f5c0a2f9a939b3a08e/src/main/java/org/jenkinsci/tools/configcloner/ConfigTransfer.java#L50

**Comment by daniel-beck** [comment]
@olivergondza To clarify, are you agreeing with me?

**Comment by pjanouse** [comment]
A assume Oliver agrees with restriction. So should I modify this PR before merging or will we go with separate JIRA&PR for restricting all already created CLI command classes?

**Comment by olivergondza** [comment]
Sorry, yes I do agree.

**Pull request merged**